File: Python3/1201 - 1400/1281 Subtract the Product and Sum of Digits of an Integer.py
# 24ms, 85.37%; 12.6MB, 100.00%
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        x, y = 1, 0
        for v in str(n):
            x, y = x * int(v), y + int(v)
        return x - y
    

File: Cpp/1201 - 1400/1281 Subtract the Product and Sum of Digits of an Integer.cpp
# 0ms, 100.00%; 8.3MB, 100.00%
class Solution
{
public:
    int subtractProductAndSum(int n)
    {
        string num = to_string(n);
        int x = 1, y = 0;
        for (auto v : num) x *= (v - 48), y += (v - 48);
        return x - y;
    }
};
